terkin/datalogger.py

```diff
@@ -310,7 +312,7 @@ def register_sensors(self):
             sensor_name = sensor_factory.__name__
             try:
                 sensor = sensor_factory()
-                if not sensor.enabled():
+                if not sensor.enabled(self.settings):
```

Comment:
Will also refactor that appropriately so that each sensor will have its settings available, regardless of system-type or environment-type.